src/main/java/net/mcreator/ui/blockly/BlocklyEditorToolbar.java

```diff
@@ -189,40 +188,44 @@ public BlocklyEditorToolbar(MCreator mcreator, String id, String extension, Bloc
 
 		add(Box.createHorizontalGlue());
 
-		JButton export = L10N.button("blockly.templates." + id + ".export");
+		JButton export = L10N.button("blockly.templates." + blocklyEditorType.getTranslationKey() + ".export");
 		export.setIcon(UIRES.get("18px.export"));
 		export.setOpaque(false);
 		add(export);
 		export.addActionListener(event -> {
-			File exp = FileDialogs.getSaveDialog(mcreator, new String[] { "." + extension });
+			File exp = FileDialogs.getSaveDialog(mcreator, new String[] { "." + blocklyEditorType.getExtension() });
 			if (exp != null) {
 				try {
-					ProcedureTemplateIO.exportAITaskSetup(blocklyPanel.getXML(), exp);
+					if (blocklyEditorType == BlocklyEditorType.AI_TASK) {
+						ProcedureTemplateIO.exportAITaskSetup(blocklyPanel.getXML(), exp);

```

Comment:
Maybe it would be better to add `start_xml_node` parameter to the `BlocklyEditorType` and pass it into this method?
I noticed that those two methods only differ by a single line, maybe you could cut them down to one?

Reply:
Sure, will do